edx_notifications/stores/sql/store_provider.py

```diff
@@ -0,0 +1,283 @@
+        if msg_type.name in self._msg_type_cache:
+            del self._msg_type_cache[msg_type.name]
+        return msg_type
+
+    def _get_notifications_for_user(self, user_id, filters=None, options=None):
+        """
+        Helper method to set up the query to get notifications for a user
+        """
+
+        # pylint/pep8 seem to complain if defaults are set to empty dicts
+        _filters = filters if filters else {}
+        _options = options if options else {}
+
+        msg_id = _filters.get('msg_id')
+        namespace = _filters.get('namespace')
+        read = _filters.get('read', True)
+        unread = _filters.get('unread', True)
+        type_name = _filters.get('type_name')
+
+        select_related = _options.get('select_related', False)
+        limit = _options.get('limit', const.MAX_NOTIFICATION_LIST_SIZE)
+        offset = _options.get('offset', 0)
+
+        # make sure passed in limit is allowed
+        # as we don't want to blow up the query too large here
+        if limit > const.MAX_NOTIFICATION_LIST_SIZE:
+            raise ValueError('Max limit is {limit}'.format(limit=limit))
+
+        if not read and not unread:
+            raise ValueError('Bad arg combination either read or unread must be set to True')
+
+        query = SQLNotificationUserMap.objects.filter(user_id=user_id)
+
+        if select_related:
+            query = query.select_related()
+
+        if msg_id:
+            query = query.filter(msg__id=msg_id)
+
+        if namespace:
+            query = query.filter(msg__namespace=namespace)
+
+        if not (read and unread):
+            if read:
+                query = query.filter(read_at__isnull=False)
+
+            if unread:
+                query = query.filter(read_at__isnull=True)
+
+        if type_name:
+            query = query.filter(msg__msg_type=type_name)
+
+        return query[offset:offset + limit]
+
+    def get_num_notifications_for_user(self, user_id, filters=None):
+        """
+        Returns an integer count of edx_notifications. It is presumed
+        that store provider implementations can make this an optimized
+        query
+
+        ARGS:
+            - user_id: The id of the user
+            - filters: a dict containing
+                - namespace: what namespace to search (defuault None)
+                - read: Whether to return read notifications (default True)
+                - unread: Whether to return unread notifications (default True)
+                - type_name: which type to return
+
+        RETURNS: integer
+
+        NOTE: You cannot pass in a filter of 'msg_id'
+        """
+
+        if filters and 'msg_id' in filters:
+            raise ValueError(
+                'It is non-sensical to pass in a msg_id '
+                'filter parameter into an aggregate counting method!'
+            )
+
+        return self._get_notifications_for_user(
+            user_id,
+            filters=filters,
+        ).count()
```

Comment:
Is this `count()` going to play well with the `limit` part of `_get_notifications_for_user`?